.github/workflows/backend_integration_tests_pr.yml

```diff
@@ -41,10 +41,14 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-buildx-
 
+      - name: Build Docker image
+        run: docker compose build
+
       - name: Run Docker Compose
         run: docker compose up -d
 
       - name: Wait for services to be up
+        timeout-minutes: 5
```

Comment:
Extra: tests were failing after 6 hours, so I've added a timeout to this infinite loop